### ISSUE TITLE
Energies for unit testing

### ISSUE THREAD

**Issue by pkozlows** [pkozlows/fci#19]
Now, I am getting a correct ground-state energy, which is certainly nice. However, I am not getting the correct full configuration interaction energy. Before I try to dive into why that is, I think it might be a good idea to make sure my condon function is outputting the right values. Do you think it would be a good idea for me to write unit tests for a pair of determinants with one and then two differences, similar to what I have done for the HF case? If so, would you be able to provide me with sample energies from your code that you have already tested?

**Comment by Walter-Feng** [comment]
Maybe you can start with <ground state | H | single excitation>. I recall it should be 0.

**Comment by pkozlows** [comment]
I probably should have thought of this before asking lol. I'm getting something like 2.220446049250313e-16, so this is correct. I imagine that <ground state | H | double excitation> won't be trivial though. I know that you probably have a lot of stuff to work on, but when you get the chance could you send this and a nontrivial single difference?

**Comment by Walter-Feng** [comment]
Um, it will then depend on how you align the determinants, so simply passing my FCI matrix may not help you with debugging, for example, my FCI matrix element of (2,1) does not directly correspond your FCI matrix element (2,1).

Or from a more egoistic perspective, I feel envious I did not receive this much help when I was writing FCI - Junjie did help me correct HF energy, he pointed out my spin was wrong, so w.r.t. these I was more than happy to help. When down to one/two electrons, it's harder to debug numerically, so I had to fight solo to nail the numbers, with only the FCI energy being the reference. 

In other words, if you make one / two electrons correct, you should get the correct energy.  So keep tweaking until you get the FCI energy. As usual, think about the range of summation, the spins, think about all the possible cases.

**Comment by pkozlows** [comment]
One thing that I have noticed while debugging is, that your code has been amazingly helpful and is written very clearly, but I often try to compare elements of my implementation, such as with the action of the handy transformer on the FCI matrix, but I'm often finding that results between our programs can be markedly different, even though we got the same final FCI energy. so, my understanding is that our matrices have to be functionally the same, but they can't be exactly the same. That is, are you able to elaborate more on what you said:

>Um, it will then depend on how you align the determinants, so simply passing my FCI matrix may not help you with debugging, for example, my FCI matrix element of (2,1) does not directly correspond your FCI matrix element (2,1).

as you have noticed, my questions often change before you are able to reply. I understand that you have better things to do, so this does not bother me at all. I hope this method of piling up my questions doesn't bother you lol

**Comment by Walter-Feng** [comment]
Say we both start on ground state, so our first determinant is the same; then in second determinant you moved electron from orbital 0 to 3, but I chose to move from 2 to 5, then the corresponding matrix elements can be different.
